open-ce/test_feedstock.py

```diff
@@ -0,0 +1,223 @@
+
+        output += "CONDA_BIN=$(dirname $(which conda))\n"
+        output += "source ${CONDA_BIN}/../etc/profile.d/conda.sh\n"
+        output += "conda activate " + self.conda_env + "\n"
+        output += self.bash_command + "\n"
+
+        return output
+
+    def run(self, conda_env_file):
+        """
+        Runs the test.
+
+        Creates a temporary bash file, and writes the contents to `get_test_command` into it.
+        Runs the generated bash file.
+        Removes the temporary bash file.
+
+        Args:
+            conda_env_file (str): The name of the original conda environment file.
+        """
+        print("Running: " + self.name)
+        # Create file containing bash commands
+        with tempfile.NamedTemporaryFile(mode='w+t', dir=os.getcwd(), delete=False) as temp:
+            temp.write("set -e\n")
+            temp.write(self.get_test_command(conda_env_file))
+            temp_file_name = temp.name
+
+        # Execute file
+        test_process = subprocess.run(["bash", temp_file_name], stdout=subprocess.PIPE, check=False,
+                                      stderr=subprocess.STDOUT, universal_newlines=True)
+
+        # Remove file containing bash commands
+        os.remove(temp_file_name)
+
+        result = TestResult(self.name, test_process.returncode, test_process.stdout)
+
+        if test_process.returncode != 0:
+            result.display_failed()
+
+        return result
+
+class TestResult():
+    """
+    Contains the results of running a test.
+
+    Args:
+        name (str): The name of the test that was run.
+        returncode (int): The return code of the test that was run.
+        output (str): The resuling output from running the test.
+    """
+    def __init__(self, name, returncode, output):
+        self.name = name
+        self.output = output
+        self.returncode = returncode
+
+    def display_failed(self):
+        """
+        Display the output from a failed test.
+        """
+        if self.failed():
+            print("--------------------------------")
+            print("Failed test: " + self.name)
+            print(self.output)
+            print("--------------------------------")
+
+    def failed(self):
+        """
+        Returns whether or not a test failed.
+        """
+        return self.returncode != 0
+
+def load_test_file(test_file):
+    """
+    Load a given test file.
+
+    Args:
+        test_file (str): Path to the test file to load.
+    """
+    with open(test_file, 'r') as stream:
+        test_file_data = yaml.safe_load(stream)
+
+    return test_file_data
+
+def gen_test_commands(test_file=utils.DEFAULT_TEST_CONFIG_FILE):
+    """
+    Generate a list of test commands from the provided test file.
+
+    Args:
+        test_file (str): Path to the test file.
+    """
+    test_data = load_test_file(test_file)
+
+    time_stamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+    conda_env = "open-ce-conda-env-" + time_stamp
```

Comment:
`open-ce-conda-env` is defined in utils.py. We can use that variable here.

Reply:
Yep, good point.